tests/TestCase.php

```diff
@@ -81,4 +84,59 @@ function (
             );
         }
     }
+
+    /**
+     * @return non-empty-list<numeric-string>
+     */
+    final protected function getAllPhpVersionsUsedByCiForTests(): array
+    {
+        $yaml = Yaml::parseFile(__DIR__.'/../.github/workflows/ci.yml');
+
+        $phpVersions = [];
+        foreach ($yaml['jobs']['tests']['strategy']['matrix']['include'] as $job) {
+            $phpVersions[] = $job['php-version'];
+        }
+
+        return array_unique($phpVersions); // @phpstan-ignore return.type (we know it's a list of parsed strings)
+    }
+
+    final protected static function getMaxPhpVersionFromEntryFile(): string
+    {
+        return self::convertPhpVersionIdToMajorMinorFormat((string) ((int) self::getPhpVersionFromEntryFileThatFollowsOperator([T_IS_GREATER_OR_EQUAL]) - 100));
+    }
+
+    final protected static function getMinPhpVersionFromEntryFile(): string
+    {
+        return self::convertPhpVersionIdToMajorMinorFormat(self::getPhpVersionFromEntryFileThatFollowsOperator('<'));
+    }
+
+    final protected static function convertPhpVersionIdToMajorMinorFormat(string $verId): string
+    {
+        $matchResult = Preg::match('/^(?<major>\d{1,2})_?(?<minor>\d{2})_?(?<patch>\d{2})$/', $verId, $capture);
+        if (!$matchResult) {
+            throw new \LogicException(\sprintf('Can\'t parse version "%s" id.', $verId));
+        }
+
+        return \sprintf('%d.%d', $capture['major'], $capture['minor']);
+    }
+
+    /**
+     * @param array{int}|string $operatorToken
+     */
+    private static function getPhpVersionFromEntryFileThatFollowsOperator($operatorToken): string
```

Comment:
I do not have the conclusion, but i'm not seeing this as best option to define supported envs as dynamic values per use-case in base test class.

i was rather thinking to define internal primitive class where we just define the constants, and reuse it in all places (except of bootstrap file).

then those getterrs like "getPhpFromEntryFile" will be only valid for entry file tests, etc. No need to populate base-class.